solus_sc/updates_view.py

```diff
@@ -372,73 +372,73 @@ def init_view(self):
 
         obsol = pisi.api.list_obsoleted()
         replc = pisi.api.list_replaces()
-        for item in sorted(upgrades):
-
-            old_item = item
-            if item in obsol:
-                if item not in replc:
-                    # No valid replacement, skip it
-                    continue
-                # Chose the replacement
-                item = replc[item][0]
-
-            new_pkg = self.packagedb.get_package(item)
-            new_version = "%s-%s" % (str(new_pkg.version),
-                                     str(new_pkg.release))
-            pkg_name = str(new_pkg.name)
-            old_pkg = None
-            systemBase = False
-
-            icon = PACKAGE_ICON_NORMAL
-            if new_pkg.partOf == "system.base":
-                systemBase = True
-
-            if self.installdb.has_package(item):
-                old_pkg = self.installdb.get_package(item)
-
-            sc_obj = ScUpdateObject(old_pkg, new_pkg)
-
-            summary = str(new_pkg.summary)
-            if len(summary) > 76:
-                summary = "%s…" % summary[0:76]
-
-            # Finally, actual size, and readable size
-            pkgSize = sc_obj.get_update_size()
-            dlSize = sc_format_size_local(pkgSize)
-
-            icon = self.appsystem.get_pixbuf_only(new_pkg)
-
-            pkg_name = self.appsystem.get_name(new_pkg)
-            summary = GLib.markup_escape_text(summary)
-
-            if old_item != item:
-                pref = "%s (replaces %s)" % (pkg_name, old_item)
-            else:
-                pref = "%s" % (pkg_name)
-
-            p_print = "%s - <small>%s</small>\n%s" % (pref,
-                                                      new_version,
-                                                      summary)
-
-            model.append(row_m, [True, False,
-                                 p_print, dlSize, icon, True, pkgSize,
-                                 sc_obj])
-
-        # Disable empty rows
-        if model.iter_n_children(row_m) == 0:
-            model.set(item, 0, False)
-            model.set(item, 1, False)
-            model.set(item, 5, False)
-
-        Gdk.threads_enter()
-        self.tview.set_model(model)
-        # Hook up events so we know what's going on (4 non blondes.)
-        self.update_from_selection()
-        model.connect_after('row-changed', self.on_model_row_changed)
-        if n_updates < 1:
-            self.stack.set_visible_child_name("check")
-        else:
+        if n_updates >= 1:
```

Comment:
Surely we could just do?

```
if n_updates == 0:
    self.stack.set_visible_child_name("check")
    Gdk.threads_leave()
    return False
```

Then keep everything else as the same indentation.

Reply:
Ah thanks, that's much better. But don't call me Surely.

![](https://i.imgur.com/fVwxl7D.jpg)